tests/show_techsupport/test_techsupport.py

```diff
@@ -0,0 +1,253 @@
+import pytest
+import os
+import pprint
+from loganalyzer import LogAnalyzer, LogAnalyzerError
+import time
+from random import randint
+from log_messages import *
+import logging
+logger = logging.getLogger(__name__)
+
+DEFAULT_LOOP_RANGE = 10
+DEFAULT_LOOP_DELAY = 10
+
+BASE_DIR = os.path.dirname(os.path.realpath(__file__))
+ACL_TMP_DIR = os.path.basename('acl_tmp')
+
+FILES_DIR = os.path.join(BASE_DIR, 'files')
+TEMPLATE_DIR = os.path.join(BASE_DIR, 'templates')
+
+ACL_RULES_FULL_TEMPLATE = 'acltb_test_rules.j2'
+ACL_REMOVE_RULES_FILE = 'acl_rules_del.json'
+ACL_RULE_PERSISTENT_FILE = 'acl_rule_persistent.json'
+ACL_RULE_PERSISTENT_DEL_FILE = 'acl_rule_persistent-del.json'
+ACL_RULE_PERSISTENT_J2 = 'acl_rule_persistent.json.j2'
+ACL_TABLE_NAME = 'DATAACL'
+
+MIRROR_RUN_DIR = os.path.basename('mirror_tmp')
+
+EVERFLOW_TABLE_NAME = "EVERFLOW"
+SESSION_INFO = {
+    'name': "test_session_1",
+    'src_ip': "1.1.1.1",
+    'dst_ip': "2.2.2.2",
+    'ttl': "1",
+    'dscp': "8",
+    'gre': "0x8949",
+    'queue': "0"
+}
+
+# ACL PART #
+
+
+def setup_acl_rules(duthost, acl_setup):
+    """
+    setup rules on DUT
+    :param dut: dut host
+    :param setup: setup information
+    :param acl_table: acl table creating fixture
+    :return:
+    """
+
+    name = ACL_TABLE_NAME
+    dut_conf_file_path = os.path.join(acl_setup['dut_tmp_dir'], 'acl_rules_{}.json'.format(name))
+
+    logger.info('generating config for ACL rules, ACL table {}'.format(name))
+    extra_vars = {
+        'acl_table_name':  name,
+    }
+    logger.info('extra variables for ACL table:\n{}'.format(pprint.pformat(extra_vars)))
+    duthost.host.options['variable_manager'].extra_vars.update(extra_vars)
+
+    duthost.template(src=os.path.join(TEMPLATE_DIR, ACL_RULES_FULL_TEMPLATE),
+                                        dest=dut_conf_file_path)
+
+    logger.info('applying {}'.format(dut_conf_file_path))
+    duthost.command('config acl update full {}'.format(dut_conf_file_path))
+
+
+@pytest.fixture(scope='function')
+def acl_setup(duthost):
+    """
+    setup fixture gathers all test required information from DUT facts and testbed
+    :param duthost: DUT host object
+    :return: dictionary with all test required information
+    """
+    logger.info('creating temporary folder for test {}'.format(ACL_TMP_DIR))
+    duthost.command("mkdir -p {}".format(ACL_TMP_DIR))
+    tmp_path = duthost.tempfile(path=ACL_TMP_DIR, state='directory', prefix='acl', suffix="")['path']
+
+    setup_information = {
+        'dut_tmp_dir': tmp_path,
+    }
+
+    logger.info('setup variables {}'.format(pprint.pformat(setup_information)))
+
+    yield setup_information
+
+
+def teardown_acl(dut, acl_setup):
+    """
+    teardown ACL rules after test by applying empty configuration
```

Comment:
Is it correct that the assumption is that there are no ACL rules on the switch? why not removing only what you added and make sure what was before the test (if any) is left?